pkg/k8s/event_handlers.go

```diff
@@ -69,6 +100,10 @@ func GetKubernetesEventHandlers(informerName, providerName string, shouldObserve
 	}
 }
 
-func getNamespace(obj interface{}) string {
-	return reflect.ValueOf(obj).Elem().FieldByName("ObjectMeta").FieldByName("Namespace").String()
+func logWithResourceName(log zerolog.Logger, obj metav1.Object) zerolog.Logger {
```

Comment:
I'd instead create  `func getNamespacedName(obj metav1.Object) string` and let the caller log something like `Observed k8s ADD event for resource 'foo/bar'`

Reply:
If we're not logging anything for ignored resources, then I can fit all the new logging stuff in a single function and make one call to that in each handler. I don't see a need to create another new function just to construct this name if it's only ever called in one place.

Reply:
Right, I didn't think we'd need a function to log, just so the log itself is inlined:
`log.Trace().Str(...,...).Msgf("... %s..", getNamespacedName(obj))`